test/jest/persistence.spec.ts

```diff
@@ -0,0 +1,155 @@
+                    }
+                    mockMPInstance._Persistence.setLocalStorage();
+                },
+            },
+            Logger: {
+                verbose: () => {},
+                warning: () => {},
+                error: () => {},
+            },
+            Identity: {
+                getCurrentUser: () => ({ getMPID: () => 'mpid' }),
+            },
+        } as unknown as IMParticleWebSDKInstance;
+
+        store = {} as IStore;
+        Store.call(store, {} as SDKInitConfig, mockMPInstance, 'apikey');
+    });
+
+    describe('#noFunctional privacy flag', () => {
+        describe('true', () => {
+            beforeEach(() => {
+                store.setNoFunctional(true);
+                store.webviewBridgeEnabled = false;
+            });
+
+            it('should NOT write to cookie when useCookieStorage is true', () => {
+                store.SDKConfig.useCookieStorage = true;
+                jest.clearAllMocks();
+
+                mockMPInstance._Persistence.update();
+
+                expect(mockMPInstance._Persistence.setCookie).not.toHaveBeenCalled();
+                expect(mockMPInstance._Persistence.setLocalStorage).not.toHaveBeenCalled();
+                expect(mockMPInstance._Persistence.setProductStorage).toHaveBeenCalled();
+            });
+
+            it('should NOT write to localStorage when useCookieStorage is false', () => {
+                store.SDKConfig.useCookieStorage = false;
+                jest.clearAllMocks();
+
+                mockMPInstance._Persistence.update();
+
+                expect(mockMPInstance._Persistence.setCookie).not.toHaveBeenCalled();
+                expect(mockMPInstance._Persistence.setLocalStorage).not.toHaveBeenCalled();
+                expect(mockMPInstance._Persistence.setProductStorage).toHaveBeenCalled();
+            });
+        });
+
+        describe('false', () => {
+            beforeEach(() => {
+                store.setNoFunctional(false);
+                store.webviewBridgeEnabled = false;
+            });
+
+            it('should write to cookie when useCookieStorage is true', () => {
+                store.SDKConfig.useCookieStorage = true;
+                jest.clearAllMocks();
+
+                mockMPInstance._Persistence.update();
+
+                expect(mockMPInstance._Persistence.setCookie).toHaveBeenCalled();
+                expect(mockMPInstance._Persistence.setLocalStorage).toHaveBeenCalled();
+                expect(mockMPInstance._Persistence.setProductStorage).not.toHaveBeenCalled();
+            });
+
+            it('should write to localStorage when useCookieStorage is false', () => {
+                store.SDKConfig.useCookieStorage = false;
+                jest.clearAllMocks();
+
+                mockMPInstance._Persistence.update();
+
+                expect(mockMPInstance._Persistence.setCookie).not.toHaveBeenCalled();
+                expect(mockMPInstance._Persistence.setLocalStorage).toHaveBeenCalled();
```

Comment:
called..but can you try to pull the item to ensure it's not null here and in other tests?